File: app/src/main/java/com/example/fruitsdiary/FruitsDiaryApplication.java
package com.example.fruitsdiary;

import android.app.Application;
import android.content.Context;

import com.example.fruitsdiary.dagger.AppComponent;
import com.example.fruitsdiary.dagger.ApplicationModule;
import com.example.fruitsdiary.dagger.DaggerAppComponent;
import com.example.fruitsdiary.dagger.NetworkModule;

public class FruitsDiaryApplication extends Application {

    private AppComponent mAppComponent;

    @Override
    public void onCreate() {
        super.onCreate();
        initAppComponent();
    }

    public AppComponent initAppComponent()
    {
        mAppComponent = DaggerAppComponent.builder()
                .applicationModule(new ApplicationModule(this))
                .networkModule(new NetworkModule())
                .build();

        return mAppComponent;
    }

    public static FruitsDiaryApplication get(Context context)
    {
        return (FruitsDiaryApplication) context.getApplicationContext();
    }

    public AppComponent getAppComponent() {
        return mAppComponent;
    }
}

File: app/src/main/java/com/example/fruitsdiary/FruitsDiaryApplication.kt
package com.example.fruitsdiary

import android.app.Application
import android.content.Context
import com.example.fruitsdiary.dagger.*

class FruitsDiaryApplication : Application() {

    val appComponent: AppComponent by lazy {
        DaggerAppComponent.builder()
                .applicationModule(ApplicationModule(this))
                .networkModule(NetworkModule())
                .presenterModule(PresenterModule())
                .build()
    }

    companion object {

        operator fun get(context: Context): FruitsDiaryApplication {
            return context.applicationContext as FruitsDiaryApplication
        }
    }
}
